### ISSUE TITLE
X3G plugin not working

### ISSUE THREAD

**Issue by LukeBonathan** [Ultimaker/Cura#4331]
I'm trying to generate X3G file using the plugin available. Every-time I do a file generates to the desktop but its corrupt and get the error message below.

I've trued changing the printer in the cura profiles and a re-install.

**3.41**

**Mac - latest os**

**Flashforge**

**generate x3g file**

**generic error message**

**x3g file to be produced**

**Additional Information
<img width="412" alt="screen shot 2018-09-06 at 13 58 31" src="https://user-images.githubusercontent.com/34811589/45159518-db020980-b1de-11e8-8a2e-f4a542e3e56f.png">
**

**Comment by nallath** [comment]
Could you also post the logs?

**Comment by LukeBonathan** [comment]
Sure, here they are from my fresh install.
[cura.log](https://github.com/Ultimaker/Cura/files/2357620/cura.log)

**Comment by nallath** [comment]
It's giving a permission denied error, which means that your operating system isn't allowing the plugin to open / use the executable it needs to actually do the conversion. What you need to do is change the permissions for the file located at 
"/Users/luke/Library/Application Support/cura/3.4/plugins/X3GWriter/X3GWriter/gpx_macos"

**Comment by LukeBonathan** [comment]
Thank you for the tip, unfortunately changing the permissions doesn't seem to work still. I applied it to the whole cura folder and still get the error.

<img width="396" alt="screen shot 2018-09-06 at 19 28 42" src="https://user-images.githubusercontent.com/34811589/45177585-482b9400-b20b-11e8-85ec-5949f6c91313.png">

**Comment by Ghostkeeper** [comment]
Next time, I suggest you report issues with this plug-in here: https://github.com/Ghostkeeper/X3GWriter/issues

After changing the permissions, did you get a different error in the log? Perhaps it's getting farther but still crashing somewhere else.

From what I can read, you can't add execute permissions from the interface in MacOS. Only with the terminal:

    sudo chmod +x gpx_macos

That's worth trying.

**Comment by LukeBonathan** [comment]
Thanks, I will do in the future. 

Here is the latest log after trying the sudo command.

[cura.log](https://github.com/Ultimaker/Cura/files/2363104/cura.log)

**Comment by diegopradogesto** [comment]
I also see this error in the log:
`2018-09-07 09:22:57,465 - ERROR - [MainThread] UM.Logger.logException [85]: FileNotFoundError: [Errno 2] No such file or directory: '/Users/luke/Library/Application Support/cura/3.4/plugins/X3GWriter/plugin.json'`

How did you install the package?

**Comment by LukeBonathan** [comment]
Thats interesting. It was installed through the cura tool box. I took a look in the folder and found the directory doubled up. See pic.

I have moved the files back removing one of the x3g folders. I now get a new error in cura...
<img width="401" alt="screen shot 2018-09-11 at 21 57 21" src="https://user-images.githubusercontent.com/34811589/45387532-3df40600-b60e-11e8-9c68-715ddbfbb384.png">
<img width="409" alt="screen shot 2018-09-11 at 22 00 59" src="https://user-images.githubusercontent.com/34811589/45387533-3df40600-b60e-11e8-936c-1a79d505ec38.png">

**Comment by LukeBonathan** [comment]
[cura.log](https://github.com/Ultimaker/Cura/files/2372731/cura.log)
Here is the latest log with that error

**Comment by TheFirstAxel** [comment]
I am still getting this error as well. Is there a fix?

**Comment by miotislucifugis** [comment]
Same here... on several Macs running different versions of OS(sierra and High Sierra) and Cura (>=3.2)

**Comment by diegopradogesto** [comment]
That is super weird. I only see these plugins being loaded:
```
2018-09-06 19:37:37,821 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [431]: Loaded plugin FileLogger
2018-09-06 19:37:37,832 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [431]: Loaded plugin LegacyProfileReader
2018-09-06 19:37:37,844 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [431]: Loaded plugin SimpleView
2018-09-06 19:37:37,856 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [431]: Loaded plugin SolidView
2018-09-06 19:37:37,869 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [431]: Loaded plugin FirmwareUpdateChecker
2018-09-06 19:37:37,881 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [431]: Loaded plugin SupportEraser
2018-09-06 19:37:37,900 - INFO - [MainThread] UM.PluginRegistry.loadPlugin [431]: Loaded plugin PrepareStage
```

and then the X3GWriter has the error that I posted before. @TheFirstAxel  and @miotislucifugis can you also post the log files?

**Comment by TheFirstAxel** [comment]
Sorry I saw that there was a version in the toolbox and updated it before I saw this. It is working now

**Comment by diegopradogesto** [comment]
So, @LukeBonathan what about you. Are you still getting the error? In that case, can you please send the entire configuration folder? (You can zip it and attach it here)

**Comment by Ghostkeeper** [comment]
I've actually posted an update for the X3GWriter at the beginning of this week. I hope it fixes a couple of things if you were using a printer without a predefined configuration for GPX, and fix the permission errors that MacOS users were getting.

The error in LukeBonathan's log is caused by Cura still registering old packages with outdated API version numbers but then not registering these plug-ins. This has been fixed in version 3.6: It should no longer show the option to save to X3G if the X3G writer is an outdated version that is no longer compatible.

So from what we've heard here, everything seems to be solved now.
